File: 0476-number-complement/0476-number-complement.cpp
class Solution {
public:
    string decimalToBinary(int &num) {
        string result = "";
        
        while(num >= 2) {
            result = to_string(num % 2) + result;
            num /= 2;
        }
        
        result = to_string(num) + result;
        return result;
    }
    
    int binaryComplementToDecimal(string input) {
        int result = 0;
        
        for (int i = 0; i < input.size(); i++) {
            if (input[input.size() - 1 - i] == '0') {
                result += (pow(2, i));
            }
        }
        
        return result;
    }
    
    int findComplement(int num) {
        return binaryComplementToDecimal(decimalToBinary(num));
    }
};

File: 0476-number-complement/0476-number-complement.kt
class Solution {

    fun getPow(input: Int, power: Int): Int {
        if (power == 0) {
            return 1
        }
        
        var result: Int = 1;
        for (i in 1..power) {
            result *= input
        }
        
        return result
    }
    
    fun decimalToBinary(nums : Int): String {
        var result: String = ""
        var num = nums
        
        while(num >= 2) {
            result = (num % 2).toString() + result
            num /= 2
        }
        
        result = num.toString() + result
        return result
    }
    
    fun binaryComplementToDecimal(input: String): Int {
        var result: Int = 0
        
        for(i in 0 until input.length) {
            if (input.get(input.length - 1 - i) == '0') {
                result += getPow(2, i)
            }
        }
        
        return result
    }
    
    fun findComplement(num: Int): Int {
        return binaryComplementToDecimal(decimalToBinary(num))
    }
}